src/Components/CreateImageWizard/steps/Packages/Packages.tsx

```diff
@@ -940,6 +968,39 @@ const Packages = () => {
   const isGroupExpanded = (group: GroupWithRepositoryInfo['name']) =>
     expandedGroups.includes(group);
 
+  const isPackageSelected = (pkg: IBPackageWithRepositoryInfo) => {
+    let isSelected = false;
+
+    if (!pkg.sources || pkg.sources[0].type === 'package') {
+      isSelected = packages.some((p) => p.name === pkg.name);
+    }
+
+    if (pkg.sources && pkg.sources[0] && pkg.sources[0].type === 'module') {
+      isSelected =
+        packages.some((p) => p.name === pkg.name) &&
+        modules.some(
+          (p) =>
+            pkg.sources && pkg.sources[0] && p.stream === pkg.sources[0].stream
+        );
+    }
+
+    return isSelected;
+  };
+
+  const isSelectDisabled = (pkg: IBPackageWithRepositoryInfo) => {
+    return (
+      (pkg.sources &&
+        pkg.sources[0].type === 'module' &&
+        modules.some(
+          (module) => pkg.sources && module.name === pkg.sources[0].name
+        ) &&
+        !modules.some(
+          (p) => pkg.sources && p.stream === pkg.sources[0].stream
+        )) ||
+      false
+    );
```

Comment:
Same here, I think a simple comment would help a lot with readability with complex conditions like this. Something like 'Disable a module package row if the stream doesn't match the stream of the already selected module'.

Reply:
I really should start commenting more, shouldn't I? Thank you, added a docstring with the conditions

Reply:
Tbh i'm generally very much on the let the code speak for itself, only add comments where really necessary. But when there are these complex multi-line conditionals, a little help is nice :D